### DISCUSSION TITLE
Return to home

### DISCUSSION THREAD

**Discussion by shiva-abba** [visose/Robots#249] [Q&A] [answered]
I intend to have an ABB robot connected to tool that holds a pen that draws several unconnected lines with one code. Currently, the code starts at a home point, begins drawing all the lines, connects them and does not return to the home point once each segment is completed. As a result, the entire drawing becomes one connected line. I need the robot to draw one line, return to home, draw another line, return to home, and so on. I could try creating a set of lines in the "negative space" of the drawing that are set on a different plane so they act as the "home" and the robot picks up the pen before drawing the next line. However, my lines are too complicated to manually do this. Is there a way to make the robot return to the home point after completing each individual line?
Thanks
![Screenshot 2023-09-14 205831](https://github.com/visose/Robots/assets/145078414/34f65e20-7bdb-4e46-9621-332274357e4c)
![image](https://github.com/visose/Robots/assets/145078414/7b1035b2-66d9-45cd-bc71-e88e533fe721)

**Comment by visose** [comment]
Why does it need to go to "home" after every line? Why not just lift the pen from the surface a little bit when moving from the end of a line to the start of the next? Alternatively, for faster drawing, you could use an actuator on the pen holder, connected to a digital signal, to retract/release the pen to drawing position.

**Reply by shiva-abba** [reply]
Thanks for your reply.
Yes, it doesn't necessarily need to go home, simply lift off the surface perpendicular to the table at the end of each line. However, I'm not sure how to do this between the two different assigned targets which dictate the home point and the movement from end point to end point to make the line.

**Reply by visose** [reply]
Check the samples folder for some help on this, but this seems more like a general Grasshopper question.

**Reply by shiva-abba** [reply]
Understood, thanks.

**Comment by robin-gdwl** [comment, marked as answer]
Adding a Retract to a List of planes is rather straightforward.
![image](https://github.com/visose/Robots/assets/25942644/4b6b62a3-f5e2-49b2-beb1-631cc149765c)
You simply take the first and last plane of each curve and move it in z-direction. 
Afterwards you insert these two planes back into the list of planes and create targets from each plane. 

see this gh-definition for reference: 
[DrawFlatCurves03.zip](https://github.com/visose/Robots/files/12682902/DrawFlatCurves03.zip)
